octopus_sensing/questionnaire/questionnaire.py

```diff
@@ -0,0 +1,123 @@
+# This file is part of Octopus Sensing <https://octopus-sensing.nastaran-saffar.me/>
+# Copyright © Nastaran Saffaryazdi 2020
+#
+# Octopus Sensing is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software Foundation,
+#  either version 3 of the License, or (at your option) any later version.
+#
+# Octopus Sensing is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with Foobar.
+# If not, see <https://www.gnu.org/licenses/>.
+
+import os
+
+from typing import List
+import csv
+import gi
+gi.require_version('Gtk', '3.0')  # nopep8
+from gi.repository import Gtk  # nopep8
+
+from octopus_sensing.questionnaire.question import Question  # nopep8
+
+
+class Questionnaire(Gtk.Window):
+    def __init__(self, name: str, experiment_id: str, stimulus_id: str,
+                 title: str, width: int = 400, height: int = 200):
+        '''
+        @param str name: Questionnaire name
+        @param str experiment_id: experiment ID
+        @param str stimulus_id: stimulus ID
+        @param str title: Questionnaire title
+
+        @kwargs int width: window's width
+        @note width: default value is 400
+        @kwargs int height: window's height
+        @note height: default value is 200
+        '''
+        Gtk.Window.__init__(self, title=title)
+        self.set_border_width(10)
+        self.set_default_size(width, height)
+        self._name = name
+        self._questions: List[Question] = []
+        self._experiment_id = experiment_id
+        self.stimulus_id = stimulus_id
+        self._output_path = "output/self_report"
+
+    def add_question(self, question: Question) -> None:
+        '''
+        Adds a question to the questionnaire
+        @param Question questions: a questions
+        '''
+        if question.id in self._questions:
```

Comment:
It's a good idea to have an assertion here to check if `question` is of type `Question`.